lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -515,9 +515,22 @@ newDBLayer trace defaultFieldValues fp = do
             selectWallet wid >>= \case
                 Nothing -> pure $ Left $ ErrNoSuchWallet wid
                 Just _ -> do
-                    Right . fmap (delegationDiscoveredFromEntity . entityVal)
-                        . take 2
+                    latest <- fmap entityVal . take 1
                         <$> selectList [CertWalletId ==. wid] [Desc CertSlot]
+                    case latest of
+                        [(DelegationCertificate _ (W.SlotId epoch _) _)] ->
```

Comment:
here, I should probably rely on current epoch not epochNo from last discovered certificate as it could be the one in the last epochs